components/batch-system/src/router.rs

```diff
@@ -118,16 +122,41 @@ where
     where
         F: FnMut(&BasicMailbox<N>) -> Option<R>,
     {
+        let maybe_clean_cache_leak = |alive_cnt, caches: &mut LruCache<u64, BasicMailbox<N>>| {
+            let leak = self.leak_count(alive_cnt);
+            for _ in 0..leak {
+                // Maybe the cache is hold leaked mailbox, so we need to remove it.
+                let _ = caches.remove_lru();
+            }
```

Comment:
question: how do we know the the leaked mailbox is going to be removed?

Reply:
No, we don't. It does best effort, should clean up leak mailbox eventually and is much faster than before.